### DISCUSSION TITLE
Enforcing taxonomic tagging with Azure Policy

### DISCUSSION THREAD

**Discussion by FallenHoot** [Azure/Enterprise-Scale#592] [Ideas]
**Describe the solution you'd like**
Tags are a crucial part of organizing your Azure resources. When following [tagging best practices within CAF](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/naming-and-tagging#naming-and-tagging-resources)

Should enforcing taxonomic with Azure Policy be added to the UI?

Examples:
Application Name
Environment
DR Tier
Cost Center
Data Classification

Inherit a tag from the subscription if missing
Inherit a tag from the subscription

Inherit a tag from the resource group
Inherit a tag from the resource group if missing

**Comment by pazdedav** [comment]
Tagging is very customer specific (I remember lengthy discussions with one customer about this topic). 

What might make sense is to offer an option to:

- define tag keys in the UI (in a new blade?)
- use this "array" as a parameter for a policy definition deployed together with other definitions
- extend the File - New subscription workflow with an option (or a requirement) to populate these tag keys with specific values, so a new subscription could be created with those tags and a new policy assignment (on this sub level) could be also assigned to secure tag inheritance
